include/nil/blueprint/parser.hpp

```diff
@@ -510,6 +542,51 @@ namespace nil {
 
                         return true;
                     }
+                    case llvm::Intrinsic::assigner_fri_lin_inter: {
+                        var s = frame.scalars[inst->getOperand(0)];
+                        var y0 = frame.scalars[inst->getOperand(1)];
+                        var y1 = frame.scalars[inst->getOperand(2)];
+                        var alpha = frame.scalars[inst->getOperand(3)];
+
+                        using component_type = components::fri_lin_inter<
+                            crypto3::zk::snark::plonk_constraint_system<BlueprintFieldType, ArithmetizationParams>, BlueprintFieldType>;
+
+                        component_type component_instance({0, 1, 2, 3, 4}, {}, {});
+
+                        components::generate_circuit(component_instance, circuits[currProverIdx], assignments[currProverIdx], {s, y0, y1, alpha}, start_row);
```

Comment:
Let's move circuit generation to separate handler to conform with other components.

Reply:
ok